src/mon/MgrMonitor.cc

```diff
@@ -546,7 +546,8 @@ bool MgrMonitor::prepare_beacon(MonOpRequestRef op)
       pending_map.available = m->get_available();
       updated = true;
     }
-    if (pending_map.available_modules != m->get_available_modules()) {
+    if (pending_map.available_modules != m->get_available_modules() &&
+        m->get_available_modules().size() > 0) {
```

Comment:
I wonder whether there couldn't be a clash between `0` due to the repetition avoidance and a really weird case of `et_available_modules()` genuinely emptied.